org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -211,6 +218,14 @@ public static class LanguageServerDocumentExecutor extends LanguageServers<Langu
 			return this.document;
 		}
 
+		public VersionedEdits toVersionedEdits(List<? extends TextEdit> edits) {
+			return new VersionedEdits(startVersion, edits, document);
+		}
+
+		public <T> Versioned<T> toVersioned(T t) {
+			return new Versioned<>(startVersion, t);
+		}
+
```

Comment:
I would move this methods to `VersionedEdits` and `Versioned` as factory methods that accept the executor?

Reply:
I think it works better (is more compact when used) as instance methods on the executor - a static factory would need two arguments, one for the result and one for the executor.

So
```
		var ex = LanguageServers.forDocument(doc).withCapability(ServerCapabilities::getDocumentFormattingProvider);
		
		var result = ex.computeFirst(ls -> ls.getTextDocumentService().formatting(params).thenApply(ex::toVersionedEdits));
```
would become
```
		var result = ex.computeFirst(ls -> ls.getTextDocumentService().formatting(params).thenApply(edits -> VersionedEdits.toVersionedEdits(edits, ex));
```
I think it is neater with the method reference rather than the lambda?

Reply:
I believe in the end we need to have version checks for each request type, as recommended in 
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#implementationConsiderations
where it recommends to check for outdated response in general (that is, for any response).

So yes, I understand that the code gets slight more verbose, but what do we do when we need versioned request for semantic highlight? Add it to the executor as well? And when we need it for hover? Add it there as well? We would end up with one method in the `LanguageServer` for each possible request type, right?

I would prefer to have those more local to the operations that use each request type, together with the subclasses of the `Versioned<T>` class.

Reply:
Ah, I see. I guess it depends on whether `Versioned<T>` (which is after all, just generic so can associated any result data with an initial timestamp) is likely to need more specialised subclasses which would be specific to the consuming class.

I think `VersionedEdits` is a special case that is likely to be all that is required for most requests that result in edits.

As a compromise, I'll create factory methods as you suggest but keep the `toVersionedEdits` on `LanguageServerDocumentExecutor` as a convenience. It's ok to have syntactic sugar for the common case, right? ;-)

Reply:
I am not sure it will be much used, but fine for me :).